applications/irda/irda-app.cpp

```diff
@@ -48,6 +48,18 @@ int32_t IrdaApp::run(void* args) {
     return 0;
 };
 
+IrdaApp::IrdaApp() {
+    furi_check(IrdaAppRemoteManager::max_button_name_length < get_text_store_size());
+    notification = static_cast<NotificationApp*>(furi_record_open("notification"));
+    irda_worker = irda_worker_alloc();
+}
+
+IrdaApp::~IrdaApp() {
+    irda_worker_free(irda_worker);
+    furi_record_close("notification");
+    for(auto& it : scenes) delete it.second;
```

Comment:
For range-based for loops you can use structured binding, like this:
```cpp
for(auto&[key,scene] : scenes)
    delete scene;
```
Also, it should be considerable to use the smart pointers there.

Reply:
Didn't know that staff. Thank you for reviewing!